### ISSUE TITLE
Simplify ssh key usage with whilp/ssh-agent and yaml syntax

### ISSUE THREAD

**Issue by rosskevin** [alienfast/docker-rails#19]
See http://stackoverflow.com/questions/32897709/ssh-agent-forwarding-inside-docker-compose-container/32911979#32911979

Similar to gems, this uses a long running volume.  Perhaps we can inject an interpolated single `ssh-agent` container into the yaml for use, with a single command, then clean it up afterwards?

**Comment by rosskevin** [comment]
Do something similar to #20 

``` yaml
# Make the host user's id_rsa key available to the web container e.g. for cloning from github
ssh-agent:
  containers:
    - web
  keys:
    - id_rsa
```

Should generate a seed default config of 

``` yaml
compose:
  web:
    environment:
      # make ssh keys available via ssh forwarding (see volume entry)
      - SSH_AUTH_SOCK=/root/.ssh/socket

    volumes_from:
      # Use configured whilp/ssh-agent long running container for keys
      - ssh-agent
```

It will also create the container and populate it with keys, prior to `up`.  I think we need to use a project_name based name for the ssh container though, and clean it up, unless we want our keys in a volume sticking around...

**Comment by rosskevin** [comment]
Unknown hosts are still a problem (anything from bitbucket or github).  Both of these fail with "Host key verification failed"

```
ssh -o 'StrictHostKeyChecking no' -T git@github.com
ssh -o 'StrictHostKeyChecking no' -T git@bitbucket.org
```

But these succeed

```
ssh -o 'StrictHostKeyChecking no' -T git@github.com
ssh -o 'StrictHostKeyChecking no' -T git@bitbucket.org
```

Need to figure out a way to either share `ssh/config` with an exclusion or `known_hosts`.

**Comment by rosskevin** [comment]
As of https://github.com/alienfast/docker-rails/commit/daea2fcbc351fdffde965d7dafb006345a99bb9a, this issue is solved, requires merging (or local building) of https://github.com/whilp/ssh-agent/pull/3

**Comment by rosskevin** [comment]
Fixed in 0.7.1
